### PR TITLE
Edit role basic UI updated

### PR DESCRIPTION
**Pull request by Achintha444**
### Purpose
Refactored the code of the basic section of the edit role UI.

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+type BasicRoleProps = IdentifiableComponentInterface & RoleEditSectionsInterface;
+
+const FORM_ID: string = "edit-role-basic";
+
+/**
+ * Component to edit basic role details.
+ *
+ * @param props - Role object containing details which needs to be edited.
+ */
+export const BasicRoleDetails: FunctionComponent<BasicRoleProps> = (props: BasicRoleProps): ReactElement => {
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx:45]: We already have a component with the same name. That will be replaced with this in the future?

File: apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        };
+
+        if (values.roleName?.toString().trim().length >= 3) {
+
+            const roleName: string = values.roleName?.toString().trim();
+
+            const searchData: SearchRoleInterface = {
+                filter: "displayName eq " + roleName,
+                schemas: [
+                    "urn:ietf:params:scim:api:messages:2.0:SearchRequest"
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx:105]: Shall we add this to constants?

File: apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+                        "fetchRoles.genericError.message")
+                }));
+            }
+            
+            return errors;
+        }
+    };
+
+    /**
+     * Method to update role name for the selected role.
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx:134]:
```suggestion
     * Update role name for the selected role.
```

File: apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx (hunk 1 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx:45]: Yes, that will be replaced with this in a future PR.

File: apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx (hunk 3 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4224/commits/9d912516de37cde05e2b5793fa9765e358586871

File: apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx (hunk 2 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4224/commits/9d912516de37cde05e2b5793fa9765e358586871

File: apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+            }
+            {
+                showRoleDeleteConfirmation && (
+                    <ConfirmationModal
+                        onClose={ (): void => setShowDeleteConfirmationModal(false) }
+                        type="negative"
+                        open={ showRoleDeleteConfirmation }
+                        assertionHint={ t("console:manage.features.roles.edit.basics.confirmation.assertionHint") }
+                        assertionType="checkbox"
+                        primaryAction="Confirm"
```
Comment (dasuni-30) [inline apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx:241]: Can we add i18n here?

**Review by dasuni-30 — APPROVED** [review]

File: apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx (hunk 4 above)
Comment (Achintha444) [reply apps/console/src/features/roles/components/edit-role/edit-role-basic-updated.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/4224/commits/7ef98d2bba165eaa30ca8d2d644b624a57acfbed

**Review by dasuni-30 — APPROVED** [review]

**Pull request merged**